### DISCUSSION TITLE
Maven classifiers with dashes gets split in classifer & extension.

### DISCUSSION THREAD

**Discussion by glennlaursen** [community/community#49682] [Packages]
### Select Topic Area

Bug

### Body

When uploading a Jar with a classifier with dashes eg., jar-with-dependencies, GitHub Packages puts 'jar' in classifier and 'with-dependencies.jar' as extension.

Here is a snip of the maven-metadata.xml for this version of the package.
```
<snapshotVersion>
  <extension>with-dependencies.jar</extension>
  <classifier>jar</classifier>
  <value>0.1.1-20230308.154437-2</value>
  <updated>20230308154455</updated>
</snapshotVersion>
```

If you don't have dashes in the classifier ie., jarwithdependecies, then it gets put correctly.

**Comment by tisonkun** [comment]
+1

This can wildy affect libraries that use os-maven-plugin to release JARs for different platform, like https://github.com/apache/incubator-opendal/pull/2335.

The classfier is like `osx-aarch_64` and I confirm that it's malformed:

```xml
         <snapshotVersion>
            <extension>x86_64.jar</extension>
            <classifier>osx</classifier>
            <value>0.1.0-20230526.123613-1</value>
            <updated>20230526123627</updated>
         </snapshotVersion>
```
